src/main/java/Duke/DukeException.java

```diff
@@ -0,0 +1,26 @@
+package Duke;
+
+/**
+ * A Parent class for Exceptions thrown in Duke.
+ */
+public class DukeException extends Exception {
+
+    /**
+     * A public constructor for Duke Exception.
+     * @param str The error message that is passed from the main code.
+     */
+    public DukeException(String str) {
+        // calling the constructor of parent Exception
+        super(str);
+    }
+
+
+    /**
+     * Returns the string representation of Duke Exception.
```

Comment:
I feel that this may be optional as it adds limited value to readers.

From [guideline](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#do-not-repeat-the-obvious):
> If the code is self-explanatory, refrain from repeating the description in a comment just for the sake of 'good documentation'.